packages/driver/test/cypress/integration/commands/assertions_spec.js

```diff
@@ -2698,6 +2724,19 @@ describe('src/cy/commands/assertions', () => {
 
         expect({}).to.have.css('foo')
       })
+
+      // https://github.com/cypress-io/cypress/issues/7353
+      it('throws when the type of the expected value is not string', (done) => {
+        cy.on('fail', (err) => {
+          expect(err.message).to.eq('TypeError: expected {} to be a string')
+
+          done()
+        })
+
+        const $div = cy.$$('<div style="display: none; position: absolute;">div</div>')
+
+        expect($div).to.have.css({})
```

Comment:
Why define a custom HTML element here? Couldn't you just use `this.$div`?

Reply:
because it doesn't depend on the fixture having the right HTML/CSS. puts all the tests dependencies inline, which is generally easier to maintain.